pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamMessageDecoder.java

```diff
@@ -46,9 +47,25 @@ public interface StreamMessageDecoder<T> {
    * @param topicName Topic name of the stream
    * @throws Exception If an error occurs
    */
+  @Deprecated
   void init(Map<String, String> props, Set<String> fieldsToRead, String topicName)
       throws Exception;
 
+  /**
+   * Initializes the decoder. This is the new interface that should be implemented by all decoders.
+   *
+   * @param props Decoder properties extracted from the {@link StreamConfig}
+   * @param fieldsToRead The fields to read from the source stream. If blank, reads all fields (only for AVRO/JSON
+   *                     currently)
+   * @param topicName Topic name of the stream
+   * @param tableSchema Pinot schema of the table
+   * @throws Exception If an error occurs
+   */
+  default void init(Map<String, String> props, Set<String> fieldsToRead, String topicName, Schema tableSchema)
```

Comment:
Let's just pass in `void init(StreamConfig streamConfig, TableConfig tableConfig, Schema schema)` since everything is extracted from them.